testlapack/dgeqp3.go

```diff
@@ -0,0 +1,152 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testlapack
+
+import (
+	"math"
+	"math/rand"
+	"sort"
+	"testing"
+
+	"github.com/gonum/blas"
+	"github.com/gonum/blas/blas64"
+	"github.com/gonum/floats"
+)
+
+type Dgeqp3er interface {
+	Dgeqp3(m, n int, a []float64, lda int, jpvt []int, tau, work []float64, lwork int)
+}
+
+func Dgeqp3Test(t *testing.T, impl Dgeqp3er) {
+	rnd := rand.New(rand.NewSource(1))
+	for c, test := range []struct {
+		m, n, lda int
+	}{
+		{1, 1, 0},
+		{2, 2, 0},
+		{3, 2, 0},
+		{2, 3, 0},
+		{1, 12, 0},
+		{2, 6, 0},
+		{3, 4, 0},
+		{4, 3, 0},
+		{6, 2, 0},
+		{12, 1, 0},
+		{1, 1, 20},
+		{2, 2, 20},
+		{3, 2, 20},
+		{2, 3, 20},
+		{1, 12, 20},
+		{2, 6, 20},
+		{3, 4, 20},
+		{4, 3, 20},
+		{6, 2, 20},
+		{12, 1, 20},
+		{129, 256, 0},
+		{256, 129, 0},
+		{129, 256, 266},
+		{256, 129, 266},
+	} {
+		n := test.n
+		m := test.m
+		lda := test.lda
+		if lda == 0 {
+			lda = test.n
+		}
+		for _, free := range []bool{false, true} {
+			a := make([]float64, m*lda)
+			for i := range a {
+				a[i] = rnd.Float64()
+			}
+			aCopy := make([]float64, len(a))
+			copy(aCopy, a)
+			jpvt := make([]int, n)
+			for j := range jpvt {
+				if free {
+					jpvt[j] = -1
+				} else {
+					jpvt[j] = j
```

Comment:
According to the docs and the code the actual value should not matter, so making it a constant or random value should work. I would change this especially if you are not going to mix free and fixed columns (I would mix them).